presto-main/src/main/java/io/prestosql/sql/planner/DomainTranslator.java

```diff
@@ -857,6 +867,88 @@ protected ExtractionResult visitBetweenPredicate(BetweenPredicate node, Boolean
                     new ComparisonExpression(LESS_THAN_OR_EQUAL, node.getValue(), node.getMax())), complement);
         }
 
+        @Override
+        protected ExtractionResult visitLikePredicate(LikePredicate node, Boolean complement)
+        {
+            Optional<ExtractionResult> result = tryVisitLikePredicate(node, complement);
+            if (result.isPresent()) {
+                return result.get();
+            }
+            return super.visitLikePredicate(node, complement);
+        }
+
+        private Optional<ExtractionResult> tryVisitLikePredicate(LikePredicate node, Boolean complement)
+        {
+            if (!(node.getValue() instanceof SymbolReference)) {
+                // LIKE not on a symbol
+                return Optional.empty();
+            }
+
+            if (!(node.getPattern() instanceof StringLiteral)) {
+                // dyamic pattern
+                return Optional.empty();
+            }
+
+            if (node.getEscape().isPresent() && !(node.getEscape().get() instanceof StringLiteral)) {
+                // dynamic escape
+                return Optional.empty();
+            }
+
+            Type type = typeAnalyzer.getType(session, types, node.getValue());
+            if (!(type instanceof VarcharType)) {
+                // not varchar
+                return Optional.empty();
+            }
+            VarcharType varcharType = (VarcharType) type;
+
+            Symbol symbol = Symbol.from(node.getValue());
+            Slice pattern = ((StringLiteral) node.getPattern()).getSlice();
+            Optional<Slice> escape = node.getEscape()
+                    .map(StringLiteral.class::cast)
+                    .map(StringLiteral::getSlice);
+
+            int patternConstantPrefixBytes = LikeFunctions.patternConstantPrefixBytes(pattern, escape);
+            if (patternConstantPrefixBytes == pattern.length()) {
```

Comment:
This should never really happen. Like expressions with constant patterns are turned into comparison expressions when ExpressionInterpreter is used to optimize expressions.

Reply:
Yes, I know. At the same time, it didn't feel right to fail or give up here.
I will add a comment.